src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx

```diff
@@ -1,16 +1,31 @@
 import React from 'react'
+import get from 'lodash/get'
+
 import { translate } from 'cozy-ui/transpiled/react/I18n'
-import toolbarContainer from 'drive/web/modules/drive/Toolbar/toolbar'
+import { withClient, models, useAppLinkWithStoreFallback } from 'cozy-client'
+import { generateUniversalLink } from 'cozy-ui/transpiled/react/AppLinker/native'
 
+import toolbarContainer from 'drive/web/modules/drive/Toolbar/toolbar'
 import styles from 'drive/styles/toolbar.styl'
-import { withClient, models, useAppLinkWithStoreFallback } from 'cozy-client'
+import useCapabilities from 'lib/hooks/useCapabilities'
 
 const CreateNoteItem = ({ client, t, displayedFolder }) => {
   const { fetchStatus, url, isInstalled } = useAppLinkWithStoreFallback(
     'notes',
     client
   )
+  const capabilities = useCapabilities(client)
+  const isFlatDomain = get(
+    capabilities,
+    'capabilities.attributes.flat_subdomains'
+  )
 
+  const returnUrl = generateUniversalLink({
+    slug: 'drive',
+    cozyUrl: client.getStackClient().uri,
+    subDomainType: isFlatDomain ? 'flat' : 'nested',
+    nativePath: `/files/${displayedFolder ? displayedFolder.id : null}`
+  })
```

Comment:
Could you add a test for this 😗 ?

Reply:
I'm not convinced by this test, I just write a defensive code, and I don't know if it worth it

Reply:
If you really want this test, I'll try to take time to write few others for the toolbar.